src/common/transformations/src/transformations/smart_reshape/lstm_states_broadcast.cpp

```diff
@@ -0,0 +1,181 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <ngraph/pass/manager.hpp>
+#include <openvino/op/util/sub_graph_base.hpp>
+#include <openvino/opsets/opset9.hpp>
+#include <transformations/smart_reshape/lstm_states_broadcast.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "dimension_tracker.hpp"
+#include "itt.hpp"
+
+ov::Input<ov::Node> get_outer_input_of_ti_by_parameter(const std::shared_ptr<ov::opset9::Parameter>& parameter,
+                                                       const std::shared_ptr<ov::opset9::TensorIterator>& ti) {
+    const auto& body = ti->get_body();
+    OPENVINO_ASSERT(body != nullptr, "TI returns invalid body graph ", ti);
```

Comment:
this is not a type of problem to hide. this would eventually fail, so let's fail timely. also I've seen that we have assert in at least one transformation.

Reply:
this is not a resposibility of the transformation to check this and throw exception
I believe it should fail in validate_and_infertype of TI op 

Transformations shouldn't affect the execution. If some transformation fails, it should simply be skipped